integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/BaseVisitorFactory.java

```diff
@@ -68,6 +69,7 @@
         new OutputDatasetWithMetadataVisitor(
             new InsertIntoHiveTableVisitor(sqlContext.sparkContext())));
     list.add(new OutputDatasetVisitor(new InsertIntoHiveDirVisitor()));
+    list.add(new OutputDatasetVisitor(new KafkaWriterVisitor()));
```

Comment:
We need to check that `KafkaSourceProvider` is on the classpath before adding this. Follow the `BigQueryNodeVisitor` example

Reply:
I guess we should do same thing for HIVE, cause we don't check it either

Reply:
@collado-mike we could generalize this in `QueryPlanVisitor` interface. We'll have similar pattern for different providers.

Reply:
Yeah, good idea. We can do that in a separate issue.